### PR TITLE
[hotfix][build] Bump lowest supported flink version to 1.17.0

### PR DESCRIPTION
**Pull request by leonardBang**
Bump lowest supported flink version to 1.17.0 as the discussion thread[1]

[1] https://lists.apache.org/thread/2qqd59nng50k4po1dlyoxkwjmvb2fmb4

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@leonardBang If we want to bump the minimum version to 1.17.0, you'll also need to address the dependency convergence.

However, looking at the code, we actually should be OK with releasing a newer version of the connector that includes support for 1.18, 1.17 and we could also still support 1.16 since the API is still compatible. It's just that we need to have a minimum of 2 Flink versions to be supported, but we can add an additional one if we want to.

**Comment by leonardBang** [comment]
> @leonardBang If we want to bump the minimum version to 1.17.0, you'll also need to address the dependency convergence.
> 
> However, looking at the code, we actually should be OK with releasing a newer version of the connector that includes support for 1.18, 1.17 and we could also still support 1.16 since the API is still compatible. It's just that we need to have a minimum of 2 Flink versions to be supported, but we can add an additional one if we want to.

Let's try to address dependency convergence first

File: .github/workflows/weekly.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
     if: github.repository_owner == 'apache'
     strategy:
       matrix:
-        flink: [1.16-SNAPSHOT, 1.17-SNAPSHOT]
+        flink: [1.17.1, 1.18.0]
```
Comment (dannycranmer) [inline .github/workflows/weekly.yml:29]: Ideally the nightly/weekly should build against 1.17-SNAPSHOT, 1.18-SNAPSHOT and 1.19-SNAPSHOT to detect breaking changes from Flink, like it is in master. Although not sure this branch weekly file is even used

Comment (MartijnVisser) [reply .github/workflows/weekly.yml:29]: > Although not sure this branch weekly file is even used

You can only trigger cron scheduled jobs from everything that's in `main`, so it doesn't matter.

**Comment by leonardBang** [comment]
Hello, @dannycranmer @MartijnVisser
 Any more comments on this PR？

**Review by dannycranmer — APPROVED** [review]
LGTM, thanks @leonardBang

**Pull request merged**